docs/rules/rules.md

```diff
@@ -2,285 +2,414 @@
-    navigation of the app.
-*   Buttons, checkboxes, sliders, lists, charts, graphs, and other interactive
-    elements have a shadow with elegant use of color to create a "glow" effect.
+* **UI Design:** Build beautiful and intuitive user interfaces that follow
+  modern design guidelines.
+* **Responsiveness:** Ensure the app is mobile responsive and adapts to
+  different screen sizes, working perfectly on mobile and web.
+* **Navigation:** If there are multiple pages for the user to interact with,
+  provide an intuitive and easy navigation bar or controls.
+* **Typography:** Stress and emphasize font sizes to ease understanding, e.g.,
+  hero text, section headlines, list headlines, keywords in paragraphs.
+* **Background:** Apply subtle noise texture to the main background to add a
+  premium, tactile feel.
+* **Shadows:** Multi-layered drop shadows create a strong sense of depth; cards
+  have a soft, deep shadow to look "lifted."
+* **Icons:** Incorporate icons to enhance the user’s understanding and the
+  logical navigation of the app.
+* **Interactive Elements:** Buttons, checkboxes, sliders, lists, charts, graphs,
+  and other interactive elements have a shadow with elegant use of color to
+  create a "glow" effect.
 
 ### Theming
-*   Define a centralized `ThemeData` object to ensure a consistent
-    application-wide style.
-*   Use Material 3 by setting `useMaterial3: true` in your `ThemeData`.
-*   Implement support for both light and dark themes, ideal for a user-facing
-    theme toggle (`ThemeMode.light`, `ThemeMode.dark`, `ThemeMode.system`).
-*   Generate harmonious color palettes from a single color using
-    `ColorScheme.fromSeed`.
-
-    ```dart
-    final ThemeData lightTheme = ThemeData(
-      useMaterial3: true,
-      colorScheme: ColorScheme.fromSeed(
-        seedColor: Colors.deepPurple,
-        brightness: Brightness.light,
-      ),
-      // ... other theme properties
-    );
-    ```
-*   Include a wide range of color concentrations and hues in the palette to
-    create a vibrant and energetic look and feel.
-*   Use specific theme properties (e.g., `appBarTheme`, `elevatedButtonTheme`)
-    to customize the appearance of individual Material components.
-*   For custom fonts, use the `google_fonts` package. Define a `TextTheme` to
-    apply fonts consistently.
-
-    ```dart
-    // 1. Add the dependency
-    // flutter pub add google_fonts
-
-    // 2. Define a TextTheme with a custom font
-    final TextTheme appTextTheme = TextTheme(
-      displayLarge: GoogleFonts.oswald(fontSize: 57, fontWeight: FontWeight.bold),
-      titleLarge: GoogleFonts.roboto(fontSize: 22, fontWeight: FontWeight.w500),
-      bodyMedium: GoogleFonts.openSans(fontSize: 14),
-    );
-    ```
+* **Centralized Theme:** Define a centralized `ThemeData` object to ensure a
+  consistent application-wide style.
+* **Material 3:** Use Material 3 by setting `useMaterial3: true` in your
```

Comment:
I think this is deprecated.  Material 3 has been the default for years.  We can probably remove this.